### PR TITLE
Ready for inspection: skyfield

### PR DESCRIPTION
**Pull request by jochym**
The recipe is working well enough (IMHO). The tests needed to be modified from the non-standard setup and reduced but otherwise it seems fine.

### CODE REVIEW

File: recipes/skyfield/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+{% set name = "skyfield" %}
+{% set version = "0.9" %}
+{% set sha256 = "0f944793cfe63c897252c589ce29e4cc71dbafce1056d657517c02bfeb8e8a72" %}
+# sha256 is the prefered checksum -- you can get it for a file with:
+#  `openssl sha256 <file name>`.
+# You may need the openssl package, available on conda-forge
+#  `conda install openssl -c conda-forge``
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Please drop all of these comments from the example.

File: recipes/skyfield/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - py2compat.patch
+
+build:
+  skip: true  # [win64]
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Why skip 64-bit Windows only?

File: recipes/skyfield/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    - skyfield
+    - skyfield.data
+    - skyfield.tests
+  commands:
+    - nosetests -v skyfield.tests.test_functions
+    - nosetests -v skyfield.tests.test_angles
+
+about:
+  home: http://rhodesmill.org/skyfield/
+  license: MIT
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Optionally can add `license_family: MIT`.

File: recipes/skyfield/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - py2compat.patch
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Has this patch been submitted upstream?

File: recipes/skyfield/meta.yaml (hunk 2 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: sgp4 package, which is a dependence, does not pass tests under win64 (different results).
I submitted the issue upstream.

File: recipes/skyfield/meta.yaml (hunk 4 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Already merged upstream

File: recipes/skyfield/meta.yaml (hunk 3 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Done

File: recipes/skyfield/meta.yaml (hunk 1 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Done

File: recipes/skyfield/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Ok, could you please add a link to the commit/PR above the patch in the recipe as a comment? Ideally with some brief explanation of its purpose.

File: recipes/skyfield/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    Python 2 and 3 and makes it easy to generate high precision research-grade 
+    positions for planets and Earth satellites.  
+  doc_url: http://rhodesmill.org/skyfield/toc.html
+  dev_url: https://github.com/skyfielders/python-skyfield
+
+extra:
+  recipe-maintainers:
+    - jochym
+    - brandon-rhodes
+
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: nit: Please strip extra newline. One terminal newline is sufficient.

File: recipes/skyfield/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    Skyfield is a pure-Python astronomy package that is compatible with both 
+    Python 2 and 3 and makes it easy to generate high precision research-grade 
+    positions for planets and Earth satellites.  
+  doc_url: http://rhodesmill.org/skyfield/toc.html
+  dev_url: https://github.com/skyfielders/python-skyfield
+
+extra:
+  recipe-maintainers:
+    - jochym
+    - brandon-rhodes
```
Comment (jochym) [inline recipes/skyfield/meta.yaml]: He agreed at the end of this PR discussion https://github.com/skyfielders/python-skyfield/pull/77 . But I suppose it is better to wait for his confirmation here. He would be potentially helpful to make hist two packages (sgp4, skyfield) compatible with win builds.

Comment (brandon-rhodes) [inline recipes/skyfield/meta.yaml]: Yes, I am willing to be a maintainer!

File: recipes/skyfield/meta.yaml (hunk 2 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Maybe not but how we are going to test without this test dependency?
The problem is the failed test is really minor it is a least significant digit +/-1 difference. IMHO it is not a failure at all. I can block the test sgp4 and make it deploy on all platforms but the question is should we do something like that? In my opinion in this case it is justified in this case (the error is really a standard case for the floating point computations).

Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: Sounds like the test may need some fixes based on what you are saying, but operation on Windows 64 is fine. Personally I would drop this `skip` entirely, but it is up to you.

File: recipes/skyfield/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+build:
+  skip: true  # [win64]
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+    - sphinx
```
Comment (brandon-rhodes) [inline recipes/skyfield/meta.yaml]: I believe that `sphinx` is only necessary for me, when building the HTML documentation and pushing it to the web site. Normal users should not need it.

File: recipes/skyfield/meta.yaml (hunk 2 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: @brandon-rhodes would you take a look at this? Particularly at this failing test for sgp4: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.8492/job/vope4rh9s4pgvf2c
The failure is at the end of the test run (only for win64 on all python versions). 
What do you think?

File: recipes/skyfield/meta.yaml (hunk 4 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Done

File: recipes/skyfield/meta.yaml (hunk 7 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: Done

**Comment by brandon-rhodes** [comment]
I hope in the future to drop dependencies like `lxml`, which are only used in obscure one-off scripts in corners of the source repository. So hopefully we can trim the long list of dependencies in the future. But for now this looks good!

**Comment by jochym** [comment]
I am glad you like the shape of the package @brandon-rhodes. Did you have a chance to take a look at this comment above: https://github.com/conda-forge/staged-recipes/pull/1830#discussion_r88003800

File: recipes/skyfield/meta.yaml (hunk 2 above)
Comment (jochym) [inline recipes/skyfield/meta.yaml]: @jakirkham I feel uneasy about ignoring failing tests - this is a slippery slope tactics. I will try to work it out with Brandon.

File: recipes/skyfield/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # Python2 compatibility patch (works for py2 and py3).
+    # Upstream PR: https://github.com/skyfielders/python-skyfield/pull/119
+    # Merged upstream - should be dropped for the next release
+    - py2compat.patch
```
Comment (jakirkham) [inline recipes/skyfield/meta.yaml]: xref: https://github.com/skyfielders/python-skyfield/pull/119

**Comment by jakirkham** [comment]
Thanks everyone.

**Pull request merged**